apps/speed.c

```diff
@@ -4747,13 +4747,14 @@ static void multiblock_speed(const EVP_CIPHER *evp_cipher, int lengths_single,
             } else {
                 int pad;
 
-                RAND_bytes(out, 16);
-                len += 16;
-                aad[11] = (unsigned char)(len >> 8);
-                aad[12] = (unsigned char)(len);
-                pad = EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_TLS1_AAD,
-                                          EVP_AEAD_TLS1_AAD_LEN, aad);
-                ciph_success = EVP_Cipher(ctx, out, inp, len + pad);
+                if (RAND_bytes(out, 16) > 0) {
+                    len += 16;
+                    aad[11] = (unsigned char)(len >> 8);
+                    aad[12] = (unsigned char)(len);
+                    pad = EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_TLS1_AAD,
+                                              EVP_AEAD_TLS1_AAD_LEN, aad);
+                    ciph_success = EVP_Cipher(ctx, out, inp, len + pad);
+                }
```

Comment:
Instead, I would use app_bail_out(). However it is not clear to me why the RAND_bytes() call is there at all - the out buffer should be overwritten by the EVP_Cipher call anyway.

@mattcaswell any idea why the RAND_bytes() call is there?

Reply:
No, I don't see an obvious reason for it.

Reply:
In that case I wouldn't bother checking the return value, just explicitly cast to (void) as we do not want to change what is the speed command measuring to keep the results stable.

Reply:
Hmm, I wonder if it's meant to be setting `inp` rather than `out`, since - unless `OPENSSL_malloc()` initialises the memory it returns - it doesn't look like `inp` is being initialised here

Reply:
Possibly

Reply:
What should we do about this issue?

Reply:
Use app_bail_out if RAND_bytes() fails instead. And also change the output parameter of RAND_bytes() to `inp`.